TRANSFORMERS/SIGNAL_PROCESSING/FFT/FFT.py

```diff
@@ -39,26 +40,42 @@ def FFT(dc_inputs: list[DataContainer], params: dict) -> DataContainer:
     window_type: str = params["window_type"]
     real: bool = params["real_signal"]
     sample_rate: int = params["sample_rate"]  # Hz
+    display: bool = params["display"]
 
     if sample_rate <= 0:
         raise ValueError(f"Sample rate must be greater than 0")
 
     signal_value = dc.y
     x = dc.x
+    sample_spacing = 1.0 / sample_rate
 
-    if window_type == "none":
-        fourier = fft.rfft(signal_value) if real else fft.fft(signal_value)
-    else:
-        window = signal.get_window(window_type, len(signal_value))
-        fourier = (
-            fft.rfft(signal_value * window) if real else fft.fft(signal_value * window)
+    if display:
+        # x-axis
+        frequency = (
+            fft.rfftfreq(x.shape[-1], sample_spacing)
+            if real
+            else fft.fftfreq(x.shape[-1], sample_spacing)
         )
+        frequency = fft.fftshift(frequency)
 
-    fourier = fft.fftshift(fourier)
-    frequency = (
-        fft.rfftfreq(x.shape[-1], 1 / sample_rate) if real else fft.fftfreq(x.shape[-1])
-    )
-    frequency = fft.fftshift(frequency)
+        # y-axis
+        if window_type == "none":
+            fourier = fft.rfft(signal_value) if real else fft.fft(signal_value)
+        else:
+            window = signal.get_window(window_type, len(signal_value))
+            fourier = (
+                fft.rfft(signal_value * window)
+                if real
+                else fft.fft(signal_value * window)
+            )
+        fourier = fft.fftshift(fourier)
+        fourier = abs(fourier)
+    else:  # used for processing
+        fourier = fft.rfft(signal_value) if real else fft.fft(signal_value)
+        frequency = x
 
-    result = abs(fourier)
-    return DataContainer(x=frequency, y=result)
+    return (
+        DataContainer(x=frequency, y=fourier)
+        if display
+        else DataContainer(x=frequency, y={"real": fourier.real, "imag": fourier.imag})
```

Comment:
Do we currently have support for dict type in 'y' key?

Reply:
I don't think so, but I couldn't think of a better way to pass complex numbers to the next node. If you know any other alternatives please lmk